### DISCUSSION TITLE
Is it possible to SELECT into list?

### DISCUSSION THREAD

**Discussion by Admolly** [duckdb/duckdb#3797] [Q&A] [answered]
Assume I have a table like the following:

|    | col   |
|---:|:----------|
|  0 | a         |
|  1 | b         |
|  2 | c         |
|  3 | c         |

I would like to do something like the following:

`SELECT 'col' as column, list_value(SELECT col FROM table) as values`

|    | column   | values               |
|---:|:---------|:---------------------|
|  0 | col      | ['a', 'b', 'c', 'c'] |

or even

`SELECT 'col' as column, list_value(SELECT DISTINCT col FROM table) as values`

|    | column   | values               |
|---:|:---------|:---------------------|
|  0 | col      | ['a', 'b', 'c'] |

Is this currently possible in duckdb?

**Comment by lnkuiper** [comment, marked as answer]
I believe you are looking for the `list` aggregate function:
```sql
D create table tbl (col varchar);
D insert into tbl values ('a'), ('b'), ('c'), ('c');
D select 'col' as column, list(col) as values from tbl;
┌────────┬──────────────┐
│ column │    values    │
├────────┼──────────────┤
│ col    │ [a, b, c, c] │
└────────┴──────────────┘
D select 'col' as column, list(distinct col) as values from tbl;
┌────────┬───────────┐
│ column │  values   │
├────────┼───────────┤
│ col    │ [a, b, c] │
└────────┴───────────┘
```

**Comment by Admolly** [comment]
Yes, perfect @lnkuiper! That is exactly what I was looking for.
